client/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for performing pre-requisite checks on Python code
+ * snippets.
+ */
+
+tie.factory('PythonPrereqCheckService', [
+  'CodePrereqCheckResultObjectFactory',
+  'PREREQ_CHECK_TYPE_BAD_IMPORT', 'PREREQ_CHECK_TYPE_MISSING_STARTER_CODE',
+  'PYTHON_STANDARD_LIBRARIES',
+  function(
+      CodePrereqCheckResultObjectFactory,
+      PREREQ_CHECK_TYPE_BAD_IMPORT, PREREQ_CHECK_TYPE_MISSING_STARTER_CODE,
+      PYTHON_STANDARD_LIBRARIES) {
+
+    var extractTopLevelFunctions = function(starterCode) {
+      var starterCodeLines = starterCode.split('\n');
+      var topLevelFunctions = [];
+      for (var i = 0; i < starterCodeLines.length; i++) {
+        var line = starterCodeLines[i].trim();
+        if (line.startsWith("def ")) {
+          topLevelFunctions.push(line);
```

Comment:
Let's call this topLevelFunctionLines, otherwise it looks like it's storing just the function names.